utils/tzktHooks.ts

```diff
@@ -0,0 +1,152 @@
+        return;
+      }
+      const tokens: Array<tzkt_token> = await response.json();
+      // First removes what is not a DeFi token
+      const defi = tokens
+        .filter(token => {
+          return (
+            token.token.metadata != null &&
+            token.token.metadata.decimals != null &&
+            token.token.metadata.symbol != null // If this field is defined then it's a defi token
+          );
+        })
+        .map(token => {
+          const contract = token.token.contract.address;
+          const decimals: number =
+            10 ** Number.parseInt(token.token.metadata?.decimals || "0"); // Should be defined because of filter
+          const balance = Number.parseInt(token.balance) / decimals; // the balance is now a float
+          const symbol: string = token.token.metadata?.symbol as string; // Checked above to be non null
+          const thumbnailUri = token.token.metadata?.thumbnailUri;
+          let icon = "";
+          if (thumbnailUri && thumbnailUri.startsWith("ipfs://")) {
+            icon = thumbnailUri.replace("ipfs://", "https://ipfs.io/ipfs/");
+          } else if (thumbnailUri && thumbnailUri.startsWith("https://")) {
+            icon = thumbnailUri;
+          } else if (!thumbnailUri) {
+            icon = `https://services.tzkt.io/v1/avatars/${contract}`; // using Tzkt avatar API
+          } else {
+          }
+          return { contract, balance, symbol, icon };
+        })
+        .filter(token => token.balance !== 0); // We don't care about tokens with 0 balance, I prefer to filter here because the balance is converted to number
+      setDefi(defi);
+    })();
+  }, [address]);
+  return defi;
+};
+
+export type price = {
+  value: number; // A float
+  evolution: number; // evolution of XTZ price over 30 days
+};
+
+/**
+ * Returns information about the price of the XTZ
+ * - the current price
+ * - the price evolution
+ * */
+export const useTzktPrice = (currency = "usd") => {
+  const [price, setPrice] = useState<price | null>(null);
+  useEffect(() => {
+    (async () => {
+      // Let's compute the price evolution
+      const pricesResponse = await fetch(
+        `https://back.tzkt.io/v1/home?quote=${currency}`
+      );
+      if (pricesResponse.status != 200) return;
+      const pricesJson = await pricesResponse.json();
+      const prices: Array<{ date: string; value: number }> =
+        pricesJson.priceChart;
+      // let's find the oldest and the most recent date
+      prices.sort((price1, price2) => {
+        const date1 = new Date(price1.date);
+        const date2 = new Date(price2.date);
+        return date1.getTime() - date2.getTime();
+      });
+      const first = prices[0];
+      const last = prices[prices.length - 1];
+      const evolution = last.value / first.value;
+
+      // Let's get the current price
+      const headResponse = await fetch("https://back.tzkt.io/v1/head");
+      if (headResponse.status != 200) return;
```

Comment:
`!==` et `===`

Reply:
fixed in: https://github.com/marigold-dev/tzsafe-ui/pull/167/commits/d6061725be57076a19783e218f046ffbbed80438